clab/clab.go

```diff
@@ -84,7 +84,9 @@ func WithGracefulShutdown(gracefulShutdown bool) ClabOption {
 // NewContainerLab function defines a new container lab
 func NewContainerLab(opts ...ClabOption) *CLab {
 	c := &CLab{
-		Config:   new(Config),
+		Config: &Config{
+			Mgmt: new(types.MgmtNet),
```

Comment:
I wonder if we need to initialize the `Mgmt` with `new()`? Since now `Config.Mgmt` is of `*types.MgmtNet`, then won't `nil` be a default value if we do

```go
c := &CLab{
		Config: new(Config),
```

Reply:
The goal from new() is not to have it be `nil`, so that even if it's not initialized later on, its values can be accessed without panicking

Reply:
true, I missed the fact that  `new(types.MgmtNet)` is not `new(*types.MgmtNet)`